src/printlayoutlistmodel.cpp

```diff
@@ -0,0 +1,76 @@
+/***************************************************************************
+  printlayoutlistmodel.cpp
+ ---------------------
+ begin                : 20.8.2018
+ copyright            : (C) 2018 by david
+ email                : david at opengisch dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "printlayoutlistmodel.h"
+#include <qgsproject.h>
+
+PrintLayoutListModel::PrintLayoutListModel(QObject* parent)
+  : QAbstractListModel( parent )
+{
+}
+
+QHash<int, QByteArray> PrintLayoutListModel::roleNames() const
+{
+  QHash<int, QByteArray> roles = QAbstractListModel::roleNames();
+  roles[TitleRow]  = "Title";
+  roles[IndexRow] = "Index";
+
+  return roles;
+}
+
+QgsProject* PrintLayoutListModel::project() const
+{
+  return mProject;
+}
+
+void PrintLayoutListModel::setProject( QgsProject* project )
+{
+  mProject = project;
+
+  reloadModel();
+}
+
+void PrintLayoutListModel::reloadModel()
+{
+  beginResetModel();
+  mPrintLayouts.clear();
+
+  const QList< QgsPrintLayout * > layouts( mProject->layoutManager()->printLayouts() );
+
+  for ( const auto &layout : layouts )
+  {
+    mPrintLayouts.append( PrintLayout( layout->name() ) );
+  }
+  endResetModel();
+
+}
+
+int PrintLayoutListModel::rowCount( const QModelIndex& parent ) const
+{
+  Q_UNUSED( parent )
+  return mPrintLayouts.size();
```

Comment:
```
if ( !parent.isValid() )
   return layouts.size()
else
  return 0;
```

Reply:
I've seen it in the QAbstractItemModel, that it should return 0. But I don't understand why...

Reply:
models are capable of providing tree-structures. So you can call this for an item at row 1 and it will give you the number of nodes http://doc.qt.io/qt-5/qtwidgets-itemviews-simpletreemodel-example.html#design-and-concepts

Reply:
Okay, I understand. Because it request of a table the amount of nonexistent children of an entry. Thanks for clarification.